src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -214,7 +215,40 @@ public TextWriter EndWritingScope()
                 _originalWriter = null;
             }
 
-            return writer;
+            var tagHelperContentWrapperTextWriter = new TagHelperContentWrapperTextWriter(Encoding.UTF8);
+            var razorWriter = writer as RazorTextWriter;
+            if (razorWriter != null)
+            {
+                razorWriter.CopyTo(tagHelperContentWrapperTextWriter);
+            }
+            else
+            {
+                tagHelperContentWrapperTextWriter.Write(writer.ToString());
+            }
+
+            return tagHelperContentWrapperTextWriter.Content;
+        }
+
+        /// <summary>
+        /// Writes an <see cref="ITextWriterCopyable"/> to the <see cref="Output"/>.
+        /// </summary>
+        /// <param name="copyableTextWriter">Contains the data to be written.</param>
+        public void Write(ITextWriterCopyable copyableTextWriter)
+        {
+            WriteTo(Output, copyableTextWriter);
+        }
+
+        /// <summary>
+        /// Writes an <see cref="ITextWriterCopyable"/> to the <paramref name="writer"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> to which the output must be written to.</param>
+        /// <param name="copyableTextWriter">Contains the data to be written to the output.</param>
```

Comment:
- `... to which the <paramref name="copyableTextWriter"> is written.`
- remove "to the output" (consistent with above overload)